backend/src/controllers/ProjectController.ts

```diff
@@ -80,14 +90,21 @@ class ProjectController {
     private createSession: () => Promise<ClientSession>
   ) {}
 
-  async getById(id: string): Promise<ProjectDoc> {
-    const project = await this.projectModel
-      .findOne({ _id: id })
-      .populate("techs");
+  async getById(id: string): Promise<any> {
+    // const project = await this.projectModel
+    //   .findOne({ _id: id })
+    //   .populate("techs");
+    const project = await this.projectModel.aggregate([
+      { $match: { _id: new mongoose.Types.ObjectId(id) } },
+      ...createJoins(),
+      createProjection(),
+      { $limit: 1 },
+    ]);
     if (!project) {
```

Comment:
If the ID is not found, the array will be empty and of length 0

Reply:
Ok, thanks, I'll look into that